contracts/factory/TREXFactory.sol

```diff
@@ -138,8 +138,21 @@ contract TREXFactory is ITREXFactory, Ownable {
     function deployTREXSuite(string memory _salt, TokenDetails calldata _tokenDetails, ClaimDetails calldata
         _claimDetails)
     external override onlyOwner {
-        require(tokenDeployed[_salt] == address(0), 'token already deployed');
-        require((_claimDetails.issuers).length == (_claimDetails.issuerClaims).length, 'claim pattern not valid');
+        require(tokenDeployed[_salt] == address(0)
+        , 'token already deployed');
+        require((_claimDetails.issuers).length == (_claimDetails.issuerClaims).length
+        , 'claim pattern not valid');
+        require((_claimDetails.issuers).length <= 5
+        , 'max 5 claim issuers at deployment');
+        require((_claimDetails.claimTopics).length <= 5
+        , 'max 5 claim topics at deployment');
+        require((_tokenDetails.irAgents).length <= 5 && (_tokenDetails.tokenAgents).length <= 5
+        , 'max 5 agents at deployment');
+        require((_tokenDetails.complianceModules).length <= 30
+        , 'max 30 module actions at deployment');
```

Comment:
```suggestion
        require((_tokenDetails.complianceModules).length <= 25
        , 'max 25 module actions at deployment');
```

We limit 25 modules at compliance level I think.

Reply:
it is not just about the max amount of modules here, but also about calls to module functions to set them up